### ISSUE TITLE
Add a method to return a list of `Color` constants

### ISSUE THREAD

**Issue by Xrayez** [godotengine/godot-proposals#1762]
**Describe the project you are working on:**
[Goost](https://github.com/goostengine/goost) Godot Engine extension.

**Describe the problem or limitation you are having in your project:**
I want to randomize colors for debugging purposes. The problem is that randomizing colors with `randf()` is problematic in some cases, as it yields either too dark or too light colors (HSV can be used, but it still could return too similar colors):

![goost_too_similar_colors](https://user-images.githubusercontent.com/17108460/97929743-7ae16080-1d72-11eb-8319-6694795d0108.png)

The idea here is that the color constants should be distinguishable enough to ease the debugging process involving the visual aspect.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Add `Color.get_constants()` to retrieve a list of all color constants. Having such a method allows to create various utility randomization/debugging scripts such as:

```gdscript
static func rand_color():
    var color = Color()
    var constants: Dictionary = color.get_constants()
    var rand_name = constants.keys()[randi() % constants.size()]
    var rand_color = constants[rand_name]
    return rand_color
```

This way, we can have a better distribution of human-perceivable colors which are easier to identify.

The proposed solution is similar to https://stackoverflow.com/a/55334738/2564620.

If we add this, `ColorN()` method available at global scope can be removed as well: #1590.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
I propose adding `Color.get_constants()`, preferably as a static method of `Color()` (yet static methods is currently not a feature in the engine).

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
You have to manually rewrite all color constants back into a script. I want to avoid doing exactly that, because there are *a lot* of them.

**Is there a reason why this should be core and not an add-on in the asset library?:**
It's a matter of exposing what's already there in the engine, else you have to dig into the engine source and expose a list of constants yourself, not trivial.

I can implement this proposal myself.

**Comment by Xrayez** [comment]
Note to self: `color_names` became an array of name/color pairs in Godot 4.0 since godotengine/godot#43385.

Regarding the problem of "too dark/too pale/too bright" for color randomization, I've actually already tried this, but produces worse result than expected.

Instead, I ended up implementing "good enough" color randomization using HSV model as seen in goostengine/goost#36, which is close to `Color(randf(), randf(), randf())` functionality but eliminates dark colors from appearing.

I'm not completely sure about the lower bound of 0.5 for saturation and value, could be increased to 0.75, but then the range of colors would be restricted to smaller HSV space by default.

What could work better is to find a database of good colors for randomization which are pleasant to human eye, and pick each color at random from a pre-populated array (the idea behind this proposal).

That said, I think this is not crucial to implement in core for the problem I've described, but perhaps there may be other use cases for this. Feel free to close if the proposal becomes stale.

**Comment by Calinou** [comment]
> What could work better is to find a database of good colors for randomization which are pleasant to human eye, and pick each color at random from a pre-populated array (the idea behind this proposal).

Feel free to copy Tailwind's color palette to a GDScript dictionary: https://github.com/tailwindlabs/tailwindcss/blob/fd8bacd7aebfabe7295e75f21bde0fb06df10ead/colors.js

Just make sure to not copy the first `module.exports` line and the last line.

**Comment by Xrayez** [comment]
Seems like could be resolved by godotengine/godot#46378 in Godot 4.0.

But the ability to retrieve color constants is available in Godot 3.2 via Goost extension in goostengine/goost#50.
